nil/tools/sol-library/func-modifiers.mdx

```diff
@@ -0,0 +1,70 @@
+# Function modifiers
+
+The =nil; Solidity library provides several function modifiers that simplify the processing of external messages and async calls.
+
+
+## `onlyInternal` and `onlyExternal`
+
+=nil; adds the following function modifiers:
+
+* `onlyInternal`
+* `onlyExternal`
+
+If a function has the `onlyInternal` modifier, it can only be called by another smart contract regardless of where this contract is located. A function with the `onlyExternal` modifier behaves the opposite way: it must always be called via an external message (e.g., from a user).
+
+:::info
+
+It is technically possible to call a function marked with `onlyExternal`/`onlyInternal` modifier with message whose type is opposite to the type in the modifier. However, such a call will produce an error.
+
+:::
+
+Consider the below examples. Contract 1 is a basic contract for voting:
+
+```solidity showLineNumbers
+import "./nil.sol";
+
+contract Elections {
+
+    using nil for address; 
+
+    mapping(string => uint256) public votes;
+
+    function requestVote(address dst, string candidate) onlyExternal public {
+        dst.async_call(address(0), gasleft(), false, msg.value, abi.encodeWithSignature("verify_vote(string)", candidate));
+    }
+
+    function addVote(string candidate) onlyExternal public {
+        votes[candidate]++;
+    }
+
+    function verifyExternal(uint64 seqno, bytes authData, bytes callData) external view returns (bool) {
+        return true;
```

Comment:
I'm not sure we want to provide an example like this, allowing anyone to vote any number of times without any authentication and deplete the funds of the contract.